libs/features/webapp-prompt/webapp-prompt.js

```diff
@@ -35,6 +35,16 @@ const getIcon = (content) => {
   return icons.company;
 };
 
+const waitForClosedGRMThen = (loadPEP) => {
```

Comment:
instead of polling the DOM, you should listen for milo:modal:closed on the window object. 
Whent that event is triggered you could then check if .locale-modal-v2 is no longer present.

Reply:
The problem is that the global-navigation.js/webapp-prompt.js might not have loaded by the time the user gets rid of the GRM, especially on slow connections, so the event would be missed and we would have to poll anyway unless we set a global flag. I'm not sure this would be all that much more readable than what we have now. Is there a particular compunction we should have against polling for the GRM?

Reply:
you can check to see if .locale-modal-v2 is present in the DOM first and if it's available there, then add an event listening for it to close.

Reply:
in general polling is worse for performance than just listening to the right event.